cereus/src/main/java/com/cactus/adapters/WebGroceryAdapter.java

```diff
@@ -2,42 +2,262 @@
 
 import com.cactus.entities.GroceryItem;
 import com.cactus.entities.GroceryList;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import okhttp3.*;
+import okhttp3.Response;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
 
 import javax.inject.Inject;
-import java.util.List;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
 
 public class WebGroceryAdapter implements GroceryAdapter {
 
+    private final static int HTTP_OK = 200;
+    private final static int HTTP_NO_CONTENT = 204;
+    private final String STATIC_IP;
     @Inject
-    public WebGroceryAdapter() {}
+    public WebGroceryAdapter() {
+        String tempIp = "192.168.0.127"; // default to this address
+        try {
+            InputStream input = new FileInputStream("src/main/resources/network.properties");
+
+            Properties props = new Properties();
+            props.load(input);
+
+            tempIp = props.getProperty("staticIp");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        STATIC_IP = tempIp;
+    }
 
     @Override
-    public List<GroceryList> getGroceryListsByUser(long userid) {
-        return null;
+    public List<GroceryList> getGroceryListsByUser(String token) {
+        OkHttpClient client = new OkHttpClient();
+
+        HttpUrl.Builder baseUrl = new HttpUrl.Builder().scheme("http").host("192.168.0.127").port(8080);
+        HttpUrl url = baseUrl.addPathSegment("api").addPathSegment("all-lists").build();
+
+        Request request = new Request.Builder()
+                .url(url)
+                .addHeader("Authorization", token)
+                .build();
+
+        HashMap<Integer, String> groceryListNames;
+
+        try {
+            // Make request
+            Response response = client.newCall(request).execute();
+            //Parse response
+            ObjectMapper finalMapper = new ObjectMapper()
+                    .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            if (response.code() != HTTP_OK) {
+                return null;
+            }
+            groceryListNames = finalMapper.readValue(Objects.requireNonNull(response.body()).string(),
+                    new TypeReference<HashMap<Integer, String>>(){});
+
+        }catch (NullPointerException | IOException i) {
+            return null;
```

Comment:
Please print a stack trace for the exception in addition to returning for ease of debugging. Ideally each type of exception (and the code that throws it) should be in its own try block, but this is ok for now.